### PR TITLE
Delegate to claim

### PR DESCRIPTION
**Pull request by julien51**
We are adding a function to claim and immediately delegate in order to increase delegation "coverage" for the UDT airdrop.

### CODE REVIEW

**Comment by julien51** [comment]
Also test suite:

```
%  yarn run test                                                                                                                                                         delegateToClaim
yarn run v1.22.11
$ yarn compile
$ rimraf ./build/
$ waffle
$ mocha


  MerkleDistributor
    #token
      ✓ returns the token address (108ms)
    #merkleRoot
      ✓ returns the zero merkle root (93ms)
    #claim
      ✓ fails for empty proof (106ms)
      ✓ fails for invalid index (102ms)
      two account tree
        ✓ successful claim (148ms)
        ✓ transfers the token (93ms)
        ✓ must have enough to transfer (172ms)
        ✓ sets #isClaimed (118ms)
        ✓ cannot allow two claims (99ms)
        ✓ cannot claim more than once: 0 and then 1 (130ms)
        ✓ cannot claim more than once: 1 and then 0 (127ms)
        ✓ cannot claim for address other than proof
        ✓ cannot claim more than proof
        ✓ gas (51ms)
      larger tree
        ✓ claim index 4 (60ms)
        ✓ claim index 9 (53ms)
        ✓ gas (51ms)
        ✓ gas second down about 15k (103ms)
      realistic size tree
        ✓ proof verification works
        ✓ gas (60ms)
        ✓ gas deeper node (58ms)
        ✓ gas average random distribution (1532ms)
        ✓ gas average first 25 (1497ms)
        ✓ no double claims in random distribution (87ms)
      delegation
        ✓ successful delegation before claim (164ms)
        ✓ successful delegation after claim (178ms)
        ✓ successful self delegation by signature after claim (253ms)
    #delegateToClaim
      ✓ successful claim and delegate to self (135ms)
      ✓ successful claim and delegate to someone else (155ms)
      ✓ gas (86ms)
    parseBalanceMap
      ✓ check the proofs is as expected
      ✓ all claims work exactly once (258ms)


  32 passing (14s)

✨  Done in 41.93s.
```

File: test/MerkleDistributor.spec.ts [hunk 1]
```diff
@@ -311,8 +314,298 @@ describe('MerkleDistributor', () => {
...
+        // Still not delegated
+        expect(await token.delegates(wallet0.address)).to.be.equal(constants.AddressZero);
+
+        const delegatee = wallet0.address
+
+        // Prep sign
+        const nonce = 0
+        const expiry = Math.floor(new Date().getTime()/1000) + 60 * 60 // 1 hour from now!
+        // const { chainId } = await provider.getNetwork()
+        const chainId = 1 // Ganache things its #1. 🤦‍♂️
```
Comment (clemsos) [inline test/MerkleDistributor.spec.ts:390]: 😓

**Comment by clemsos** [comment]
LGTM

I have a small gas offset error while running it locally tho
```
MerkleDistributor
       #delegateToClaim
         gas:
     AssertionError: Expected "184765" to be equal 184757
      at /Users/clemsos/Dev/unlock/merkle-distributor/test/MerkleDistributor.spec.ts:972:59
      at step (test/MerkleDistributor.spec.ts:52:23)
      at Object.next (test/MerkleDistributor.spec.ts:33:53)
      at fulfilled (test/MerkleDistributor.spec.ts:24:58)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

**Comment by julien51** [comment]
yeah I found that it changes from time to time for me as well. The test computes averages and I think it may vary based on compilation and a few other things.

**Pull request merged**